src/anet.c

```diff
@@ -591,7 +607,10 @@ static int _anetTcpServer(char *err, int port, char *bindaddr, int af, int backl
         return ANET_ERR;
     }
     for (p = servinfo; p != NULL; p = p->ai_next) {
-        if ((s = socket(p->ai_family, p->ai_socktype, p->ai_protocol)) == -1) continue;
+        rv = anetTcpGetProtocol(mptcp);
+        if (rv == ANET_ERR) goto error;
+
+        if ((s = socket(p->ai_family, p->ai_socktype, rv)) == -1) continue;
```

Comment:
detail: `anetTcpGetProtocol()` no longer returns an error. You could even use `anetTcpGetProtocol()` directly if prefer:

```suggestion
        if ((s = socket(p->ai_family, p->ai_socktype, anetTcpGetProtocol(mptcp))) == -1) continue;
```

Reply:
Good point!

Reply:
@matttbe Thanks! Apply this suggestion in the latest version.